File: backend/app/schemas/intelligence.py
from decimal import Decimal

from pydantic import BaseModel


class ProfitSummaryResponse(BaseModel):
    revenue: Decimal
    cogs: Decimal
    gross_profit: Decimal
    gross_margin: Decimal


class ProductProfitabilityResponse(BaseModel):
    product_id: int
    quantity_sold: Decimal
    revenue: Decimal
    cogs: Decimal
    gross_profit: Decimal
    gross_margin: Decimal


class InventoryProductResponse(BaseModel):
    product_id: int
    product_name: str
    sku: str
    warehouse_id: int
    warehouse_name: str
    current_stock: Decimal
    reorder_level: Decimal
    stock_value: Decimal
    low_stock: bool


class InventoryIntelligenceResponse(BaseModel):
    total_stock_value: Decimal
    low_stock_products: int
    products: list[InventoryProductResponse]

File: frontend/src/types/intelligence.ts
export interface ProfitSummary {
  revenue: string;
  cogs: string;
  gross_profit: string;
  gross_margin: string;
}

export interface ProductProfitability {
  product_id: number;
  quantity_sold: string;
  revenue: string;
  cogs: string;
  gross_profit: string;
  gross_margin: string;
}

export interface InventoryProduct {
  product_id: number;
  product_name: string;
  sku: string;
  warehouse_id: number;
  warehouse_name: string;
  current_stock: string;
  reorder_level: string;
  stock_value: string;
  low_stock: boolean;
}

export interface InventoryIntelligence {
  total_stock_value: string;
  low_stock_products: number;
  products: InventoryProduct[];
}

export interface RevenueTrend {
  period: string;
  revenue: string;
  cogs: string;
  gross_profit: string;
}